WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -122,6 +126,20 @@ - (BOOL)fb_framelessFuzzyMatchesElement:(XCElementSnapshot *)snapshot
     valuesAreEqual(self.placeholderValue, snapshot.placeholderValue);
 }
 
+- (BOOL)fb_hasMoreThanOneVisibleChildSnapshot
```

Comment:
Yes, but first query asks for cells and if that fails, than you grab everything.
You can easily replace `cellSnapshots = self._allDescendants` with `[self fb_descendantsMatchingType:XCUIElementTypeAny]` and then you base only on types in given order.

Having that, you can share logic that chooses parent to scroll and logic that calculates scrolling vector. You simply pass same types and probably use same filtering block